### PR TITLE
fix: test_certificate_insert_remove

### PR DESCRIPTION
**Pull request by joske**
* fix this test
* implement Storage::remove_transmission

### CODE REVIEW

File: node/narwhal/src/helpers/storage.rs [hunk 1]
```diff
@@ -296,7 +296,10 @@ impl<N: Network> Storage<N> {
         self.certificates.write().remove(&certificate_id);
         // Remove the batch ID.
         self.batch_ids.write().remove(&batch_id);
-        // TODO (howardwu): Remove the transmissions.
+        // Remove the transmissions.
```
Comment (howardwu) [inline node/narwhal/src/helpers/storage.rs:299]: This case is not so easy.

We can only remove the transmission if this certificate is the **last** instance that references the transmission ID.

This was why it was initially left as a TODO to resolve.

Comment (joske) [reply node/narwhal/src/helpers/storage.rs:299]: Oh, then we'll just keep this as a draft for now.

**Comment by howardwu** [comment]
I've fixed it in this commit: https://github.com/AleoHQ/snarkOS/commit/16cae7a39c8a9b694fecd24e8d801a86d65b0cf4

**Comment by joske** [comment]
superseded

**Pull request closed without merging**